nexus/src/app/deployment.rs

```diff
@@ -179,7 +182,25 @@ impl super::Nexus {
             })
             .collect();
 
-        Ok(PlanningContext { creator, policy: Policy { sleds } })
+        // The choice of which inventory collection to use here is not
+        // necessarily trivial.  Inventory collections may be incomplete due to
+        // transient (or even persistent) errors.  It's not yet clear what
+        // general criteria we'll want to use in picking a collection here.  But
+        // as of this writing, this is only used for one specific case, which is
+        // to implement a gate that prevents the planner from provisioning
+        // non-NTP zones on a sled unless we know there's an NTP zone already on
+        // that sled.  For that purpose, it's okay if this collection is
```

Comment:
Are we soon going to have other uses for the latest inventory in the planner?

Reply:
It's a good question.  I am not sure yet.  We _mostly_ don't need it for zones (aside from this case) because we can always tell sled agents exactly what zones to run and it will just do nothing if it's already running those zones.  We might eventually want the inventory when we have operations that don't work that way, where it's expensive to do them and you don't want to re-do them if you don't have to.  The MGS-managed updates (like SP update, host OS flash update) come to mind, where you want to avoid doing the update if you know it's already at the right version.  We may instead want to update slot 0, wait for the inventory to reflect the new version in slot 0, then update slot 1, etc.  But at this point I think that use case is pretty far away.